internal/bgp/bgp.go

```diff
@@ -671,3 +686,28 @@ func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
 
 	return t
 }
+
+// localAddressExists returns true if the address addr exists on any of the
+// network interfaces in the ifs slice. The function doesn't return errors and
+// instead returns false if an error has occurred while looking for the
+// address. This function is safe to call if addr is nil or malformed.
+func localAddressExists(ifs []net.Interface, addr net.IP) bool {
+	for _, i := range ifs {
+		addresses, err := i.Addrs()
+		if err != nil {
+			return false
+		}
+
+		for _, a := range addresses {
+			ip, ok := a.(*net.IPNet)
+			if !ok {
+				return false
```

Comment:
a similar thing here: while this is more weird and probably never happens, it feels safer to just ignore ifaces with invalid IPs, and continue to look at the rest, instead of stop looking completely and return false.

I mean, there is no reason to not look at other iterfaces and if one of the others is well formed and all, use it, right?